src/commands/deploy/setPreDeployTaskForMavenModule.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import { TaskDefinition } from 'vscode';
+import { IDeployContext } from 'vscode-azureappservice';
+import * as constants from '../../constants';
+import { javaUtils } from "../../utils/javaUtils";
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../../vsCodeConfig/settings';
+import * as tasks from '../../vsCodeConfig/tasks';
+
+export async function setPreDeployTaskForMavenModule(context: IDeployContext, module: { pom: string, artifactId: string, packaging: string }): Promise<void> {
+    const preDeployTaskKey: string = 'preDeployTask';
+    const workspaceFspath: string = context.workspaceFolder.uri.fsPath;
```

Comment:
I think you should pass in the effectiveDeployFsPath or originalDeployFsPath rather than using the workspaceFolder.uri.fsPath here.  I ran into a bug where it was using my first workspace folder rather than the one I deployed.

Look below:

![image](https://user-images.githubusercontent.com/5290572/122343219-53eb2f80-cefa-11eb-831a-84eb41bfccc4.png)

Reply:
**This seems a bug of the `getDeployFsPath.getContainingWorkspace`** ?! should I fix it?

I tried with the same project and deploying `user-service` to Azure webapp and found that this problem occurs only when all these 3 conditions meets:
* `user-service` is a submodule of `java-microservice` and
* both `user-service` and `java-microservice` are added as root folders of your workspace and 
* `java-microservice` is before `user-service` in workspace.

![image](https://user-images.githubusercontent.com/69189193/122421848-42b82680-cfbf-11eb-992a-9d1f3d6dddea.png)
![image](https://user-images.githubusercontent.com/69189193/122423507-895a5080-cfc0-11eb-894e-9f0164cffedc.png)
the `workspaceFolder` becomes correct after changing the order of the root folders in workspace(you can drag the root folders to change their order):
![image](https://user-images.githubusercontent.com/69189193/122424175-169da500-cfc1-11eb-8a7e-000bcc06f4e8.png)

**the `context.workspaceFolder` is copied from `deployPaths: IDeployPaths` and originally generated by `getDeployFsPath.getContainingWorkspace`, which iterates the root folders and find the first one(this is exactly why the order matters) containing the selected folder as `workspaceFolder`**

Reply:
![image](https://user-images.githubusercontent.com/69189193/122428159-44381d80-cfc4-11eb-8263-077bbea4e76a.png)

Reply:
![image](https://user-images.githubusercontent.com/69189193/122428657-af81ef80-cfc4-11eb-8839-0d31a7b0f9d4.png)

Reply:
![image](https://user-images.githubusercontent.com/69189193/122428984-fd96f300-cfc4-11eb-8569-f7bc1e724ecd.png)

Reply:
No, I don't think you need to fix this bug in your PR. Do we have an issue to track it? I would guess it's an uncommon edge case to have two workspace folders where one is a sub folder of another, so I'm not super worried about it.

Reply:
issue is reported here:
https://github.com/microsoft/vscode-azureappservice/issues/2085